src/Lemon/Containers/Converters/Ncch2Binary.cs

```diff
@@ -0,0 +1,237 @@
+        {
+            if (file != null) {
+                long position = writer.Stream.Position;
+                int offset = offsetInUnits * NcchHeader.Unit;
+                int fileLength = (int)file.Stream.Length;
+                int fileLengthInUnits = fileLength / NcchHeader.Unit;
+
+                bool usedChunk = false;
+                var fileRange = Enumerable.Range(offset, fileLength);
+                foreach (var usedUnit in usedUnits) {
+                    var usedRange = Enumerable.Range(usedUnit.Key, usedUnit.Value);
+
+                    if (fileRange.Intersect(usedRange).Any()) {
+                        usedChunk = true;
+                    }
+                }
+
+                if (writer.Stream.Length > offset && usedChunk) {
+                    offset = (int)Math.Ceiling((double)writer.Stream.Length / NcchHeader.Unit) * NcchHeader.Unit;
+                    offsetInUnits = offset / NcchHeader.Unit;
+                }
+
+                if (writer.Stream.Length < offset) {
+                    long tempPosition = writer.Stream.Position;
+
+                    writer.Stream.Position = writer.Stream.Length;
+                    writer.Write(new byte[offset - writer.Stream.Length]);
+
+                    writer.Stream.Position = tempPosition;
+                }
+
+                writer.Write(offsetInUnits);
+                writer.Write(fileLengthInUnits);
+
+                writer.Stream.Position = offset;
+
+                file.Stream.WriteTo(writer.Stream);
+
+                writer.Stream.Position = position + 0x8;
+
+                usedUnits.Add(offset, fileLength);
+            }
+            else {
+                writer.Write(0);
+                writer.Write(0);
+            }
+
+            if (hasHashRegion) {
+                writer.Write(hashRegion);
+            }
+        }
+
+        void WriteSHA256(DataWriter writer, Node file, int hashRegion = 0, bool hasHashRegion = false)
+        {
+            using (SHA256 sha256 = SHA256.Create()) {
+                try {
+                    if (file == null) {
+                        writer.Write(new byte[0x20]);
+                        return;
+                    }
+
+                    if (hasHashRegion && hashRegion == 0) {
+                        writer.Write(sha256.ComputeHash(new byte[0x0]));
+                    } else if (hasHashRegion) {
+                        int hashSize = hashRegion * 0x200;
+                        byte[] buffer = new byte[hashSize];
+                        file.Stream.Position = 0;
+                        file.Stream.Read(buffer, 0, hashSize);
+
+                        writer.Write(sha256.ComputeHash(buffer));
+                    } else {
+                        writer.Write(sha256.ComputeHash(file.Stream));
+                    }
+                } catch (Exception ex) {
+                    throw new FormatException(ex.Message);
```

Comment:
To encapsulate an exception it's better to include the full `ex` variable, so we can access to its information (like stacktrace and line numbers) via the `InnerException` property of this new format exception. Something like:
```csharp
throw new FormatException($"Failed to perform SHA256 for {file.Name}", ex);
```